modules/docs/arrow-docs/docs/docs/optics/cons/README.MD

```diff
@@ -0,0 +1,53 @@
+---
+layout: docs
+title: Cons
+permalink: /docs/optics/cons/
+---
+
+## Cons
+
+{:.beginner}
+beginner
+
+`Cons` provides a [Prism]({{ '/docs/optics/prism' | relative_url }}) between a structure `S` and its first element `A` and tail `S`.
+It provides a convenient way to attach or detach elements to the beginning side of a structure [S].
+
+It can be constructed by providing the `Prism`.
+
+```kotlin:ank
+import arrow.data.ListK
+import arrow.optics.instances.listk.cons.cons
+import arrow.optics.typeclasses.Cons
+
+val listFirst = ListK.cons<Int>().cons()
+val instance = Cons(listFirst)
+instance
+```
+
+It defines two functions `cons` and `uncons`.
+
+`cons` prepends an element `A` to a structure `S`.
+
+```kotlin:ank
+import arrow.optics.instances.list.cons.cons
+
+1.cons(listOf(2, 3))
+```
+
+`uncons` detaches the first element `A` from a structure `S`.
+
+```kotlin:ank
+import arrow.optics.instances.list.cons.uncons
+
+listOf(1, 2, 3).uncons()
+```
+```kotlin:ank
+emptyList<Int>().uncons()
+```
+
+### Instances
```

Comment:
This can be auto-generated once this is approved https://github.com/arrow-kt/arrow/pull/1119